src/graph/options/chart_options.pas

```diff
@@ -0,0 +1,100 @@
+unit chart_options;
+
+{$mode objfpc}{$H+}
+
+interface
+
+uses
+  Classes, sysutils, ast, options_hashmap, ast_types, Graphics, executor;
+
+const
+  anaColors: array of TColor =
+    (clBlack, clBlue, clRed, clGreen, clYellow, clWhite, clSkyBlue, clFuchsia, clGray, clAqua);
+
+type
+
+  TColorMap = array of TColor;
+
+  function ToColors(AValue: String; out Msg: UTF8String): TColorMap;
+  function ToColors(AHexValues: Array of String; out Msg: UTF8String): TColorMap;
+  function ChartColorsFromOptions(OptionList: TOptionList; Executor: TExecutor): TColorMap;
+  function AxisTypeFromVariableType(AVar: TVariableList; AIndex: Integer): TASTResultTypes;
+implementation
+
+uses
+  LazUTF8, ana_globals, typinfo, math, strutils, options_utils;
+
+function ChartColorsFromOptions(OptionList: TOptionList; Executor: TExecutor): TColorMap;
```

Comment:
I am not a fan of having a dependency to the TExecutor in this unit when not strictly needed. It creates a coupling in the code that can be hard to remove if we need a refactor later on.

Reply:
It was a bad strategy to gain access to error messages. So removed. Errors in colours are communicated to calling routine via msg string